app/src/test/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeListViewModelTest.kt

```diff
@@ -0,0 +1,89 @@
+package nerd.tuxmobil.fahrplan.congress.changes
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import info.metadude.android.eventfahrplan.commons.testing.MainDispatcherTestRule
+import info.metadude.android.eventfahrplan.commons.testing.assertLiveData
+import info.metadude.android.eventfahrplan.commons.testing.verifyInvokedNever
+import info.metadude.android.eventfahrplan.commons.testing.verifyInvokedOnce
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.flowOf
+import nerd.tuxmobil.fahrplan.congress.NoLogging
+import nerd.tuxmobil.fahrplan.congress.TestExecutionContext
+import nerd.tuxmobil.fahrplan.congress.models.Meta
+import nerd.tuxmobil.fahrplan.congress.models.Session
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+
+class ChangeListViewModelTest {
+
+    @get:Rule
+    val mainDispatcherTestRule = MainDispatcherTestRule()
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @Test
+    fun `changeListParameter returns null`() {
+        val repository = createRepository(sessionsFlow = emptyFlow())
+        val viewModel = createViewModel(repository)
+        assertLiveData(viewModel.changeListParameter).isEqualTo(null)
+        verifyInvokedNever(repository).readMeta()
+        verifyInvokedNever(repository).readUseDeviceTimeZoneEnabled()
+    }
+
+    @Test
+    fun `changeListParameter returns zero sessions`() {
+        val repository = createRepository()
+        val viewModel = createViewModel(repository)
+        val expected = ChangeListParameter(emptyList(), 0, false)
+        assertLiveData(viewModel.changeListParameter).isEqualTo(expected)
+        verifyInvokedNever(repository).readMeta()
+        verifyInvokedNever(repository).readUseDeviceTimeZoneEnabled()
```

Comment:
Newlines at least between the several phases would be helpful while reading it.

Reply:
:x: I disagree. With a short function body I experience white space as cluttering the file more than having a compact function body.